src/services/KeywordExtractionService.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Extracts keywords of given PDF document. Keywords are
+ * extracted from metadata of the PDF or, if they are not
+ * found there, they are extracted from 'Keywords' section
+ * in the text of PDF document. If no keywords are found,
+ * an empty array is returned.
+ * @async
+ * @param {PDFDocumentProxy} pdfDoc PDF document
+ * @returns {String[]} keywords array
+ */
+async function extractKeywords(pdfDoc) {
+  let keywords = await getKeywordsFromMetadata(pdfDoc);
+
+  if (keywords.length == 0)
+    keywords = await getKeywordsFromText(pdfDoc);
+
+  console.log(keywords); //TODO: remove
+
+  return keywords;
+}
+
+/**
+ * Extracts keywords from metadata of given PDF document
+ * and returns them in array of strings. If keywords are
+ * not in the PDF document metadata an empty array is
+ * returned.
+ * @async
+ * @param {PDFDocumentProxy} pdfDoc PDF document
+ * @returns {String[]} keywords array
+ */
+async function getKeywordsFromMetadata(pdfDoc) {
+  const metadata = await pdfDoc.getMetadata();
+  
+  let keywords = [];
+  if (metadata.info.Keywords) 
+    keywords = metadata.info.Keywords.split(", ");
+  else
+    console.warn("Keywords not in metdata!");
```

Comment:
Is this neccessary, this warn?

Reply:
We need to think of a way of disabling the toggle button, or sending a message to the user that keywords could not have been extracted to use that functionallity.

Does not have to be a part of this PR, but we need something like that.

Reply:
I agree with everything you said and made changed accordingly